share/sysbuild/CMakeLists.txt

```diff
@@ -91,4 +91,25 @@ while(NOT "${images_length}" EQUAL "${processed_length}")
   set(processed_length ${processed_length_new})
 endwhile()
 
+# Include zephyr modules generated sysbuild CMake file.
+foreach(module_name ${SYSBUILD_MODULE_NAMES})
+  # Note the second, binary_dir parameter requires the added
+  # subdirectory to have its own, local cmake target(s). If not then
+  # this binary_dir is created but stays empty. Object files land in
+  # the main binary dir instead.
+  # https://cmake.org/pipermail/cmake/2019-June/069547.html
+  zephyr_string(SANITIZE TOUPPER MODULE_NAME_UPPER ${module_name})
+  if(NOT ${SYSBUILD_${MODULE_NAME_UPPER}_CMAKE_DIR} STREQUAL "")
+    set(SYSBUILD_CURRENT_MODULE_DIR ${SYSBUILD_${MODULE_NAME_UPPER}_MODULE_DIR})
+    set(SYSBUILD_CURRENT_CMAKE_DIR ${SYSBUILD_${MODULE_NAME_UPPER}_CMAKE_DIR})
+    add_subdirectory(${SYSBUILD_CURRENT_CMAKE_DIR} ${CMAKE_BINARY_DIR}/modules/${module_name})
+  endif()
+endforeach()
+# Done processing modules, clear SYSBUILD_CURRENT_MODULE_DIR and SYSBUILD_CURRENT_CMAKE_DIR.
+set(SYSBUILD_CURRENT_MODULE_DIR)
+set(SYSBUILD_CURRENT_CMAKE_DIR)
```

Comment:
Can this be moved further up? Right now, the `sysbuild.cmake` of images added here are not being picked up.

Reply:
The problem with moving this up is it would then cause problems, if you assume a custom task for signing images, running that before processing `sysbuild.cmake` files will then end up with a mismatch of images that have signing enabled and those that do not.

Reply:
Ok, then I hope you find another fix, because leaving out `sysbuild.cmake` is a problem too. Signing should be deferred until all added images have been processed.

Reply:
actually, this should be moved up.
Module code inclusion should be very early in the process, preferable around here:
https://github.com/zephyrproject-rtos/zephyr/blob/9f46db90c832e56363c5d7bb42892651b39b271d/share/sysbuild/CMakeLists.txt#L32

Reason for that is to allow modules to provide extra features, extra CMake modules, CMake adjustments or even override / extending Zephyr defined CMake functions.

Wrt. custom signing and sysbuild (as well as other custom / post-process tools), then the exact design of such downstream integration has not been completed.
Basically two paths are under consideration:
- ability by Zephyr sysbuild to specify module name(s) which should be include as post-image process.
- ability to specify extra tools by providing an infrastructure, similar to how things are done for SCA:
   Ref: https://docs.zephyrproject.org/latest/develop/sca/index.html
   Ref: #52671

I believe first option is more CMake aligned, but not an infrastructure supported in Zephyr modules today, and doesn't provide user flexibility as an SCA like infrastructure provides.
Even if an SCA like structure is provided, then it doesn't mean user must set a variable, a module itself can specify a custom tool to use, such as signing tool, by forcing the CMake variable to such tool.
Also an SCA like infrastructure could even make it possible to provide natively supported signing implementations while still provide a clean way for other users to provide alternative implementations / own custom signing support and easily specify that their tools is to be used.

Regardless of the path chosen, both methods described requires this code to be moved further up.

Reply:
Moved it up, have not moved it to line 33 as modules should have visibility of the application and its configuration so have placed it after the main application is configured